stream_audio/streamaudio.cpp

```diff
@@ -837,12 +894,25 @@ void *AudioStream::StreamCallback(int *size) {
 int AudioStream::ReadFileData(int buf, int len) {
   if (!m_archive.Opened())
     return 0;
-
-  if (m_decoder) {
-    // We have a compressed stream
-    return m_decoder->Read(m_buffer[buf].data, len);
+  
+  int big_endian = 0;
+  #if OUTRAGE_BIG_ENDIAN
+  big_endian = 1;
+  #endif
+  int len_requested = len;
+  //mprintf((0,"Reading stream: len=%d bytes_read=0,len_requested=%d\n",len,len_requested));
+  int bytes_read = acm_read(m_acm,m_buffer[buf].data,len_requested,big_endian,2,1);
```

Comment:
this might return a value < 0 on error, and 0 on EOF, I think those cases should be handled?

Reply:
Good call! Thanks!